wcfsetup/install/files/lib/system/flood/FloodControl.class.php

```diff
@@ -0,0 +1,241 @@
+<?php
+namespace wcf\system\flood;
+use wcf\data\object\type\ObjectTypeCache;
+use wcf\system\SingletonFactory;
+use wcf\system\WCF;
+use wcf\util\DateUtil;
+use wcf\util\UserUtil;
+
+/**
+ * Flood control tracks the times contents are created by users.
+ * 
+ * @author      Matthias Schmidt
+ * @copyright   2001-2020 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package     WoltLabSuite\Core\System\Flood
+ * @since       5.4
+ */
+class FloodControl extends SingletonFactory {
```

Comment:
```suggestion
final class FloodControl extends SingletonFactory {
```

Prefer making classes `final` where no clear need for extension exists. Removing the `final` is easily possible, the inverse is not.

When making the class `final` also adjust `protected` to `private`.